internal/backend/secrets/svc.go

```diff
@@ -3,36 +3,62 @@ package secrets
 import (
 	"context"
 	"fmt"
+	"time"
 
 	"github.com/lithammer/shortuuid"
 	"go.uber.org/zap"
 	"golang.org/x/exp/maps"
 
+	"go.autokitteh.dev/autokitteh/internal/backend/configset"
 	"go.autokitteh.dev/autokitteh/internal/xdg"
 	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
 )
 
-type secrets struct {
-	impl   Secrets
-	logger *zap.Logger
+type Config struct {
+	Type     string        `koanf:"type"`
+	VaultURL string        `koanf:"vault_url"`
+	Timeout  time.Duration `koanf:"timeout_duration"`
+}
+
+var Configs = configset.Set[Config]{
+	Default: &Config{
+		Timeout: 1 * time.Minute,
+	},
 }
 
-func New(l *zap.Logger) (sdkservices.Secrets, error) {
-	// TODO(ENG-145): Allow the user to select the implementation via CLI.
-	impl, err := NewFileSecrets(l, xdg.DataHomeDir())
+func New(l *zap.Logger, cfg *Config) (sdkservices.Secrets, error) {
+	var impl Secrets
+	var err error
+
+	switch cfg.Type {
```

Comment:
You can use fx to provide the relevant implementation instead of this switch condition

Wdyt?

Reply:
I really prefer to rely on explicit checking and initialization which is traceable, and less on code injection that isn't easy to trace.